### ISSUE TITLE
rake rails4:check raise NoMethodError: undefined method name for Array

### ISSUE THREAD

**Issue by lucasprag** [alindeman/rails4_upgrade#5]
Hi, I just tried this gem for the first time and I got a `NoMethodError: undefined method `name` for #<Array:0x005590f01ba248>` when running `rake rails4:check`.

I noticed that the `spec` of the [gemfile.rb](https://github.com/alindeman/rails4_upgrade/blob/master/lib/rails4_upgrade/gemfile.rb#L12) is a `Bundler::LazySpecification`. Is that right?

Should I have to do something different? 

Thank you very much for your time on help me with this =)

More information:
```bash
ruby --version
ruby 2.2.7p470 (2017-03-28 revision 58194) [x86_64-linux]

rails --version
Rails 3.2.22.4
```

**Comment by rbgrouleff** [comment]
I've run in to this error as well.

The stack trace I'm getting is this:

```
** GEM COMPATIBILITY CHECK **
rake aborted!
NoMethodError: undefined method `name' for #<Array:0x007fe5976df3b0>
/Users/rasmus/.gem/ruby/2.2.8/gems/rails4_upgrade-0.5.0/lib/rails4_upgrade/gemfile.rb:24:in `block in dependencies'
/Users/rasmus/.gem/ruby/2.2.8/gems/rails4_upgrade-0.5.0/lib/rails4_upgrade/gemfile.rb:23:in `each'
/Users/rasmus/.gem/ruby/2.2.8/gems/rails4_upgrade-0.5.0/lib/rails4_upgrade/gemfile.rb:23:in `map'
/Users/rasmus/.gem/ruby/2.2.8/gems/rails4_upgrade-0.5.0/lib/rails4_upgrade/gemfile.rb:23:in `dependencies'
/Users/rasmus/.gem/ruby/2.2.8/gems/rails4_upgrade-0.5.0/lib/rails4_upgrade/incompatible_gems.rb:10:in `incompatibilities'
/Users/rasmus/.gem/ruby/2.2.8/gems/rails4_upgrade-0.5.0/lib/rails4_upgrade/formatters/incompatible_gems_formatter.rb:11:in `output'
/Users/rasmus/.gem/ruby/2.2.8/gems/rails4_upgrade-0.5.0/lib/rails4_upgrade/tasks/upgrade.rake:15:in `block (2 levels) in <top (required)>'
/Users/rasmus/.gem/ruby/2.2.8/gems/rails4_upgrade-0.5.0/lib/rails4_upgrade/tasks/upgrade.rake:5:in `block (2 levels) in <top (required)>'
/Users/rasmus/.gem/ruby/2.2.8/gems/rake-12.3.0/exe/rake:27:in `<top (required)>'
/Users/rasmus/.gem/ruby/2.2.8/bin/bundle:23:in `load'
/Users/rasmus/.gem/ruby/2.2.8/bin/bundle:23:in `<main>'
Tasks: TOP => rails4:check_gems
```

As far as I can see, this error occurs because `@lockfile.dependencies.map` yields an arrays instead of a `Bundler::Dependency` objects in line 24 in `Rails4Upgrade::Gemfile`.

I'm btw. using Bundler v1.16.1.

Furthermore I've dug around in the Bundler source, and I can see that in Bundler v1.14.6 `Bundler::LockfileParser#dependencies` is an Array instance, but in v1.15.0 it has changed to a Hash instance.

When I change bundler version to v1.14.6, I'm able to execute the rake task successfully.

The immediate fix (I guess) is to downgrade Bundler to v1.14.6.

I've also forked this repo and created a branch that adds compatibility for Bundler versions newer than v1.14.6: https://github.com/rbgrouleff/rails4_upgrade/tree/add_compatibility_for_newer_bundler_versions

@alindeman Please let me know, if you'd like me to create a pull request.

**Comment by alindeman** [comment]
> @alindeman Please let me know, if you'd like me to create a pull request.

@rbgrouleff Yes please, I'd accept that PR, especially if it worked with both Bundler version ranges.
